### PR TITLE
Replaced connection error by Retry error when getting manifest for

### PR DESCRIPTION
**Pull request by midnightercz**
registry source

### CODE REVIEW

File: src/pushsource/_impl/backend/registry_source.py [hunk 1]
```diff
@@ -108,10 +108,9 @@ def _push_item_from_registry_uri(self, uri, signing_key):
                         manifest_types=[mtype],
                     )
                     self._manifests[source_uri][mtype] = manifest_details
-                except (
-                    requests.exceptions.ConnectionError,
-                    requests.exceptions.HTTPError,
-                ) as e:
+                except requests.exceptions.RetryError as e:
```
Comment (rohanpm) [inline src/pushsource/_impl/backend/registry_source.py:111]: What's the reasoning behind this?  RetryError is what's raised after all retries are exhausted, it's not clear why this should be ignored.

Could you point at a specific problem (i.e. with traceback) which would be solved by this?

Comment (midnightercz) [reply src/pushsource/_impl/backend/registry_source.py:111]: Instead of HTTPError, RetryError is raise even on 404. But 404 should be tolerated here.

Comment (rohanpm) [reply src/pushsource/_impl/backend/registry_source.py:111]: It looks OK now.  In the version of the code where I posted this comment, you didn't check for 404 but instead caught all kinds of retryerror with no filtering.

File: test-requirements-py2.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 jinja2
 pytest
 mock
-rpm-py-installer
+rpm-py-installer < 4.16
```
Comment (rohanpm) [inline test-requirements-py2.txt:4]: If you need this then please do it in its own PR with the reasoning explained.

Comment (midnightercz) [reply test-requirements-py2.txt:4]: I accidentally included it in this commit

**Review by chandwanitulsi — APPROVED** [review]
